src/remote.c

```diff
@@ -1541,6 +1541,20 @@ static int opportunistic_updates(const git_remote *remote, const git_remote_call
 	return error;
 }
 
+static int truncate_fetch_head(const char *gitdir)
+{
+	git_buf path = GIT_BUF_INIT;
+	int error;
+
+	if ((error = git_buf_joinpath(&path, gitdir, GIT_FETCH_HEAD_FILE)) < 0)
+		return error;
+
+	error = git_futils_truncate(path.ptr, 0666);
```

Comment:
Shouldn't this be `0644`?

Reply:
No, it should be `0666` and the user's `umask` will apply.  However, we should be using `GIT_REFS_FILE_MODE`.

Reply:
Changed.